lib/mozart_fetcher/application.ex

```diff
@@ -43,6 +43,10 @@ defmodule MozartFetcher.Application do
     # See https://hexdocs.pm/elixir/Supervisor.html
     # for other strategies and supported options
     opts = [strategy: :one_for_one, name: MozartFetcher.Supervisor]
-    Supervisor.start_link(children, opts)
+    Supervisor.start_link(children ++ hackney_setup(), opts)
+  end
+
+  defp hackney_setup do
+    [:hackney_pool.child_spec(:origin_pool, timeout: 1000, max_connections: 20000)]
```

Comment:
Probably for fetcher `20,000` is way too much

Reply:
also the timeout should be hardcoded?

Reply:
hmm, fetcher makes far more HTTP requests than Ingress per request it receives? I agree it is too many, but perhaps so is Ingress then?

Reply:
I'll use `Application.get_env(:mozart_fetcher, :timeout)` to get the timeout

Reply:
agree, way too much, but also in that in this case we would risk overflowing of requests our origin.

Reply:
Have you seen this? https://github.com/bbc/mozart-fetcher/blob/master/lib/mozart_fetcher/http_client.ex#L7

Reply:
so are we OK to set the `timeout`, and `recv_timeout` to the same value? This might be misleading as our combined timeout will be the sum of them both?

Useful explanation of the difference: https://github.com/edgurgel/httpoison/issues/211#issuecomment-271968830

Reply:
`timeout` is the time we use to create a connection - means that the client shouldn't wait that much in case of faults. 
So `timeout` should be much smaller than `recv_timeout`